### PR TITLE
Add support for current CentOS releases

### PR DESCRIPTION
**Pull request by smortex**
This allows us to run the test suite against EL.

This PR also include:

* #15 
* #18 
* #17 
* #19 
* #20

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -6,9 +6,17 @@
...
-      # FIXME: https://github.com/puppetlabs/puppetlabs-apt/pull/1015
-      ensure_packages('apt-transport-https')
-      Package['apt-transport-https'] -> Class['apt::update']
+      if fact('os.family') == 'debian' {
+        # FIXME: https://github.com/puppetlabs/puppetlabs-apt/pull/1015
+        ensure_packages('apt-transport-https')
+        Package['apt-transport-https'] -> Class['apt::update']
+      }
+
+      if fact('os.family') == 'redhat' {
```
Comment (faxm0dem) [inline spec/acceptance/class_spec.rb]: Shouldn't this be `== 'RedHat'` ?
Same comment for L9

**Review by faxm0dem — CHANGES_REQUESTED** [review]

File: spec/acceptance/class_spec.rb (hunk 1 above)
Comment (smortex) [reply spec/acceptance/class_spec.rb]: :+1:  for consistency with the rest of the module.  String comparisons are case insensitive for ASCII chars with puppet:
https://puppet.com/docs/puppet/7/lang_expressions.html#lang_exp_comparison_operators

Comment (smortex) [reply spec/acceptance/class_spec.rb]: Fixed

**Review by faxm0dem — APPROVED** [review]
LGTM !

**Pull request merged**